cep-XXXX.md

```diff
@@ -0,0 +1,291 @@
+
+### Preprocessing selectors
+
+`environment.yml` files MAY specify per-line filters that contain an expression that MUST be
+evaluate to a boolean. If `true`, the expression MUST be removed and the value MUST be kept in the
+document. If `false`, both the expression and the value MUST be removed.
+
+#### Comment-based selectors
+
+The syntax for these selectors MUST be in the form of a line comment with the expression between
+square brackets: `# [<expression>]`. The space between `#` and the opening square bracket is
+optional but recommended. Once the selectors have been processed, the resulting document MUST be
+parsable as YAML.
+
+The expressions MUST be composed of one or more boolean variables, and can be chained with `and`
+and `or` logic operators. Precedence can be specified with parentheses.
+
+The supported boolean variables are the OS- and architecture-specific values collected in
+[conda-build's 25.1.x documentation][selectors], plus `osx64` (equivalent to `osx and x64`). More
+specifically,`py`, `py*`, `np` and `build_platform` are NOT supported.
+
+These are used by `conda-lock`.
+
+#### Dictionary-based selectors
+
+The `dependencies` section MAY contain elements that are `dict[str, str]` instead of `str`. The key
+in these cases MUST be of the form `sel(<expression>)`, where `<expression>` evaluates to a
+boolean. The expression MUST only contain one of the following boolean variables: `unix` (true when
+the target OS is Linux or macOS), `linux`, `osx`, or `win` (true when the target OS is Linux, macOS
+or Windows, respectively).
+
+The value MUST be a `MatchSpec`-compatible string. When the expression is true, the dictionary
+entry in the list MUST be replaced with the value.
+
+These are used by `micromamba`.
+
+## Examples
+
+Simplest possible `environment.yml`:
+
+```yaml
+dependencies:
+  - numpy
+```
+
+With a name:
+
+```yaml
+name: test
+dependencies:
+  - numpy >=1.10
+```
+
+With channels:
+
+```yaml
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - numpy
+```
+
+With a `pip` section:
+
+```yaml
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - numpy
+  - pip:
+      - scipy
+```
+
+With variables:
+
+```yaml
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - numpy
+variables:
+  MY_ENV_VAR: "My Value"
+```
+
+With platforms:
+
+```yaml
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - numpy
+platforms:
+  - linux-64
+```
+
+With a category:
+
+```yaml
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - pytest
+category: test
+```
+
+With a comment-based selector:
+
+```yaml
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - python
+  - pywin32  # [win]
+```
+
+With a dictionary-based selector:
+
+```yaml
+name: test
+channels:
+  - conda-forge
+dependencies:
+  - python
+  - sel(win): pywin32
+```
+
+## Reference
+
+- [Allowed extensions](https://github.com/conda/conda/blob/24.5.0/conda/env/specs/yaml_file.py)
```

Comment:
Is this referring to extensions as in `.yml` and `.yaml`?
In that case, maybe specify the [line](https://github.com/conda/conda/blob/24.5.0/conda/env/specs/yaml_file.py#L11) instead as in the following references below?